### DISCUSSION TITLE
Task 2: Position of closest scalar

### DISCUSSION THREAD

**Discussion by rsbgonzalez** [rramosp/ai4eng.v1#18] [02 - PYTHON]
Profe buenas noches

Estoy intentando hacer esta task, pero no entiendo el ejemplo:

Given a 1D vector  x , find the position of the closest element to  v:

Lo que entiendo de lo que se pretende es que dado un vector de n elementos, se debe encontrar la posición del elemento más cercano a V. En el ejemplo:

>> x=np.arange(25,55,3)
>> minimo(x,34)
3

No entiendo porque la solución es 3, si:
- La interpretación es de posiciones, la posición3 no existe.
- Si la interpretación es del valo más cercano a V, en este caso 34, la solución debería ser la posición de 25, es decir, 0 (cero).

Por favor me indica cuál es mi mal interpretación?

**Comment by luthors** [comment]
Al parecer tengo tu misma mal interpretación, si ya lo solucionaste me podrías colaborar ....
muchas gracias

**Reply by rsbgonzalez** [reply]
Hola @luthors ... no aún no he podido, estoy pendiente de que el profe @rramosp @raulramosudea pueda verlo

**Reply by luthors** [reply]
Creo que ya encontré porque tenemos la mala perspectiva, ésta línea x=np.arange(25,55,3), no crea un vector de 3 elementos, realmente crea este arreglo array([25, 28, 31, 34, 37, 40, 43, 46, 49, 52]), entonces ahí si aplica el índice  3 que arroja en el resultado

**Reply by rramosp** [reply]
exacto luthors!!!!!! buena investigación ... verdaderamente hay que mirar en las tripas de las cosas para entenderlas ... no dejéis que el python os domine ... sois vosotros los que tenéis que dominar la herramienta

**Reply by rsbgonzalez** [reply]
@rramosp buenas tardes,

Escribí mi código, pero no entiendo porque cuando lo corro  me funciona, y cuando lo envío y el sistema lo testea no funciona.
Este es mi código:

def minimo(x,v):
  x = np.arange(25,55,3)
  v = 76
  a = v - min(x) 
  b = len(x)
  c = round((max(x)-min(x))/b)
  c1 = round(a/c)
  return (c1)

**Reply by rsbgonzalez** [reply]
@Felipe-RA@github será que me puedes ayudar por favor?

**Comment by rramosp** [comment]
por qué v=76 dentro de la función??? y por qué también le pones un valor a
x???

El lun, 31 ene 2022 a las 14:19, rsbgonzalez ***@***.***>)
escribió:

> @rramosp <https://github.com/rramosp> buenas tardes,
>
> Escribí mi código, pero no entiendo porque cuando lo corro me funciona, y
> cuando lo envío y el sistema lo testea no funciona.
> Este es mi código:
>
> def minimo(x,v):
> x = np.arange(25,55,3)
> v = 76
> a = v - min(x)
> b = len(x)
> c = round((max(x)-min(x))/b)
> c1 = round(a/c)
> return (c1)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/18#discussioncomment-2084073>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIOF4ORCPI224JHOO6LUY3ODZANCNFSM453XCMTQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Reply by rsbgonzalez** [reply]
Profe para verificar que el código funcionara, pero ya se los quité y me funcionó. Muchas gracias!!
